src/sys/stat.rs

```diff
@@ -236,6 +357,134 @@ pub fn umask(mode: Mode) -> Mode {
     Mode::from_bits(prev).expect("[BUG] umask returned invalid Mode")
 }
 
+/// File metadata.
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct FileStat(libc::stat);
+
+impl From<libc::stat> for FileStat {
+    fn from(stat: libc::stat) -> Self {
+        FileStat(stat)
+    }
+}
+
+impl FileStat {
+    /// Returns the ID of device containing this file
+    pub fn dev(&self) -> type_of_st_dev {
+        self.0.st_dev
+    }
+
+    /// Returns the Inode number
+    pub fn ino(&self) -> type_of_st_ino {
+        self.0.st_ino
+    }
+
+    /// Returns the number of hard links
+    pub fn nlink(&self) -> type_of_st_nlink {
+        self.0.st_nlink
+    }
+
+    /// Returns a number encoding `file type` and `permission`
+    pub fn mode(&self) -> type_of_st_mode {
+        self.0.st_mode
+    }
+
+    /// Returns the User ID of the file owner
+    pub fn uid(&self) -> Uid {
+        Uid::from_raw(self.0.st_uid)
+    }
+
+    /// Returns the Group ID of the file owner
+    pub fn gid(&self) -> Gid {
+        Gid::from_raw(self.0.st_gid)
+    }
+
+    /// Returns the device ID (if this is a special file)
+    pub fn rdev(&self) -> type_of_st_rdev {
+        self.0.st_rdev
+    }
+
+    /// Returns the total size, in bytes
+    pub fn size(&self) -> type_of_st_size {
+        self.0.st_size
+    }
+
+    /// Returns the block size
+    pub fn blksize(&self) -> type_of_st_blksize {
+        self.0.st_blksize
+    }
+
+    /// Returns the number of blocks allocated
+    pub fn blocks(&self) -> type_of_st_blocks {
+        self.0.st_blocks
+    }
+
+    /// Returns the time of last access
+    pub fn atime(&self) -> TimeSpec {
+        #[cfg(target_os = "netbsd")]
+        return TimeSpec::new(self.0.st_atime, self.0.st_atimensec);
+        #[cfg(not(target_os = "netbsd"))]
+        return TimeSpec::new(self.0.st_atime, self.0.st_atime_nsec);
+    }
+
+    /// Returns the time of last modification
+    pub fn mtime(&self) -> TimeSpec {
+        #[cfg(target_os = "netbsd")]
+        return TimeSpec::new(self.0.st_mtime, self.0.st_mtimensec);
+        #[cfg(not(target_os = "netbsd"))]
+        return TimeSpec::new(self.0.st_mtime, self.0.st_mtime_nsec);
+    }
+
+    /// Returns the time of last status change
+    pub fn ctime(&self) -> TimeSpec {
+        #[cfg(target_os = "netbsd")]
+        return TimeSpec::new(self.0.st_ctime, self.0.st_ctimensec);
+        #[cfg(not(target_os = "netbsd"))]
+        return TimeSpec::new(self.0.st_ctime, self.0.st_ctime_nsec);
+    }
+
+    #[cfg(any(
+        target_os = "openbsd",
+        target_os = "macos",
+        target_os = "ios",
+        target_os = "freebsd",
+        target_os = "dragonfly",
+        target_os = "netbsd",
+    ))]
+    /// Return the file flags
+    pub fn flags(&self) -> FileFlag {
+        FileFlag::from_bits(self.0.st_flags as _).unwrap()
```

Comment:
This `as _` cast is necessary as on `FreeBSD/DragonFlyBSD/NetBSD`, `st_flags` and `type_of_st_flag` (`type_of_file_flag`) have different types.